test/cache.py

```diff
@@ -59,6 +59,16 @@ def test_lmod_conf(self):
             'dir': './test/data',
         })
 
+    def test_versions(self):
+        data = [
+            [
+                ['v2106-foss-2021a-highopt', '8-foss-2020b-20210316', '8-intel-2020b', '8-foss-2020b', 'v2106-foss-2021a', '9-foss-2021a'],
+                ['v2106-foss-2021a-highopt', 'v2106-foss-2021a', '9-foss-2021a', '8-intel-2020b', '8-foss-2020b-20210316', '8-foss-2020b'],
```

Comment:
I do not grok the sorting here, especially not the sorting of the two last items.

Reply:
pick any order, it's in any case much better than the current raised exception ;)